### PR TITLE
BaSP-MR-125:Fix email regex and added custom messages for other regex

### PR DESCRIPTION
**Pull request by CavalloFede**
Changed the email regex for a new one
Added custom messages for the regex validation

### CODE REVIEW

**Review by Gianluca27 — APPROVED** [review]

**Review by leo-balonchard — APPROVED** [review]
Readyyy

File: src/validations/member.js [hunk 1]
```diff
@@ -3,22 +3,36 @@ import Joi from 'joi';
...
+        'string.pattern.base': 'Last name must be only made of letters(it can be a compound Name)',
+      })
       .required(),
     dni: Joi.number().min(1000000).max(99999999).integer()
       .required(),
     phone: Joi.number().min(1000000000).max(9999999999).integer()
       .required(),
-    email: Joi.string().regex(/^[^@]+@[^@]+.[a-zA-Z]{2,}$/).min(5).max(30)
+    email: Joi.string().regex(/^(?!\.)(?!.*\.\.)[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(\.[a-z-]+)*(\.[a-z]{2,4})$/).min(5).messages({
+      'string.pattern.base': 'Email is not valid, muy contain only one @ and a valid domain',
```
Comment (Micaela-Rossi) [inline src/validations/member.js:20]: Mistake in "muy"

File: src/validations/member.js [hunk 2]
```diff
@@ -32,18 +46,32 @@ const validateCreate = (req, res, next) => {
...
-      .regex(/^[^@]+@[^@]+.[a-zA-Z]{2,}$/)
+    email: Joi.string().regex(/^(?!\.)(?!.*\.\.)[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(\.[a-z-]+)*(\.[a-z]{2,4})$/).min(5)
+      .messages({
+        'string.pattern.base': 'Email is not valid, muy contain only one @ and a valid domain',
+      })
       .lowercase(),
-    city: Joi.string().min(2).max(20).regex(/^[a-zA-Z]+(?:[\s-][A-Za-z]+)*$/),
+    city: Joi.string().min(2).max(20).regex(/^[a-zA-Z]+(?:[\s-][A-Za-z]+)*$/)
+      .messages({
+        'string.pattern.base': 'Password must contain at least 8 characters and cannot contain blank spaces',
```
Comment (Micaela-Rossi) [inline src/validations/member.js:66]: this is for password, this is city message
Captura de pantalla 2023-06-13 104602

**Review by Micaela-Rossi — COMMENTED** [review]
In general therms everything seems  working fine. Just little comments

**Pull request merged**